### DISCUSSION TITLE
Integration with TypeDoc

### DISCUSSION THREAD

**Discussion by TooTallNate** [fuma-nama/fumadocs#526] [Q&A] [answered]
I'm curious if and how are people integrating TypeDoc output into Fumadocs?

I am currently looking into using [`typedoc-plugin-markdown`](https://typedoc-plugin-markdown.org) for this, but wondering if there's any better solutions.

**Comment by fuma-nama** [comment]
Fumadocs comes with a Typescript integration which generates docs from type definitions, you can refer to the docs. If you want more features, welcome to open a feature request

**Reply by TooTallNate** [reply]
Oh right, I forgot about that. I was evaluating using it, but I'm having trouble wrapping my head around it. I have a monorepo with a few packages, and I want to have separate pages for every exported item. Do I need a `.model.mdx` file for every single type/class/interface/etc.? How would I go about doing that in an automated way?

**Reply by fuma-nama** [reply]
Unfortunately yes, I'm not aiming to make a generator that generates the entire docs based on typedoc, and the currently API is mainly built for the official docs site itself. It's useful for smaller use cases like generating docs for UI components (with auto type table), but you might want to use the plugin you have mentioned (or other parts of the ecosystem) if they suit you better.

You can suggest some features too, but has to be more specific (like what API exactly do you need)

**Reply by fuma-nama** [reply]
Also I've documented the `generateMDX` API, you can pass the MDX content instead of specifying them via files.

**Comment by TooTallNate** [comment, marked as answer]
I ended up getting to a TypeDoc + `typedoc-plugin-markdown` configuration that I am satisfied with ([example](https://nxjs.n8.io/runtime/api/functions/fetch)).

One thing that would be nice would be the ability to disable the TOC links up to a certain level, i.e. I don't want TOC links for `<h3>` or greater to remove all these "Returns" links:

<img width="154" alt="Screenshot 2024-06-30 at 03 13 19" src="https://github.com/fuma-nama/fumadocs/assets/71256/d845438c-81e3-47d9-9e0c-bc9739f5222f">

**Reply by fuma-nama** [reply]
In fact you can easily do this from your page.tsx, filter the toc items before passing it to the page component

**Reply by TooTallNate** [reply]
Ahh right, ya makes sense. Thanks, I'll get that handled.
